src/directives/schema-validate.js

```diff
@@ -1,110 +1,173 @@
-          // Angular 1.2
-          ngModel.$setViewValue(ngModel.$viewValue);
+        // Get in last of the parses so the parsed value has the correct type.
+        // We don't use $validators since we like to set different errors depeding tv4 error codes
+        ngModel.$parsers.push(validate);
+
+        // Listen to an event so we can validate the input on request
+        scope.$on('schemaFormValidate', function() {
+          if (ngModel.$setDirty) {
+            // Angular 1.3+
+            ngModel.$setDirty();
+            validate(ngModel.$modelValue);
+          } else {
+            // Angular 1.2
+            ngModel.$setViewValue(ngModel.$viewValue);
+          }
+
+        });
+
+
+        var DEFAULT_DESTROY_STRATEGY;
+        if (scope.options && scope.options.formDefaults) {
+          var formDefaultDestroyStrategy = scope.options.formDefaults.destroyStrategy;
+          var isValidFormDefaultDestroyStrategy = (formDefaultDestroyStrategy === undefined ||
+          formDefaultDestroyStrategy === '' ||
+          formDefaultDestroyStrategy === null ||
+          formDefaultDestroyStrategy === 'retain');
+          if (isValidFormDefaultDestroyStrategy) {
+            DEFAULT_DESTROY_STRATEGY = formDefaultDestroyStrategy;
+          }
+          else {
+            console.warn('Unrecognized formDefaults.destroyStrategy: \'%s\'. Used undefined instead.',
+                formDefaultDestroyStrategy);
+            DEFAULT_DESTROY_STRATEGY = undefined;
+          }
         }
 
-      });
+        // Clean up the model when the corresponding form field is $destroy-ed.
+        // Default behavior can be supplied as a formDefault, and behavior can be overridden in the form definition.
+        scope.$on('$destroy', function() {
+          var form = getForm();
+          var destroyStrategy = form.destroyStrategy; // Either set in form definition, or as part of formDefaults.
+          var schemaType = getSchemaType();
+
+          if (destroyStrategy && destroyStrategy !== 'retain' ) {
+            // Don't recognize the strategy, so give a warning.
+            console.warn('Unrecognized destroyStrategy: \'%s\'. Used default instead.', destroyStrategy);
+            destroyStrategy = DEFAULT_DESTROY_STRATEGY;
+          }
+          else if (schemaType !== 'string' && destroyStrategy === '') {
+            // Only 'string' type fields can have an empty string value as a valid option.
+            console.warn('Attempted to use empty string destroyStrategy on non-string form type. Used default instead.');
+            destroyStrategy = DEFAULT_DESTROY_STRATEGY;
+          }
+
+          if (destroyStrategy === 'retain') {
+            return; // Valid option to avoid destroying data in the model.
+          }
+
+          destroyUsingStrategy(destroyStrategy);
+
+          function destroyUsingStrategy(strategy) {
+            var strategyIsDefined = (strategy === null || strategy === '' || typeof strategy == undefined);
```

Comment:
Uhmmm... shouldn't `typeof strategy == undefined`  be `typeof strategy === 'undefined'` or  `strategy === undefined` or `angular.isUndefined(strategy)